app/src/firebase/cups/CupService.ts

```diff
@@ -6,6 +6,22 @@ export async function createNewCup(cup: Cup) {
     await firebase.firestore().collection("cups").add(cup);
 }
 
+export async function getCurrentCupIfExists(): Promise<Cup | undefined> {
+    const querySnapshot = await firebase
+        .firestore()
+        .collection("cups")
+        .get({ source: "server" });
+    const publishedDocQuery = querySnapshot.docs.find((doc) => {
+        return !doc.data().isPublished;
+    });
+    if (publishedDocQuery) {
+        const publishedDoc = publishedDocQuery.data();
+        publishedDoc.id = publishedDocQuery.id;
+        return publishedDoc as Cup;
+    }
+    return undefined;
+}
+
```

Comment:
I would prefer to stream the list of all unpublished, open cups from the db.
That should return a list containing 0 or 1 cups (currently).
That will make us more a little more future proof.